app/src/main/java/net/gsantner/markor/util/DocumentIO.java

```diff
@@ -139,16 +169,29 @@ public static synchronized boolean saveDocument(final Document document, final S
                 //noinspection ResultOfMethodCallIgnored
                 document.getFile().getParentFile().mkdirs();
             }
-            if (shareUtil.isUnderStorageAccessFolder(document.getFile())) {
-                shareUtil.writeFile(document.getFile(), false, (fileOpened, fos) -> {
-                    try {
-                        fos.write(document.getContent().getBytes());
-                    } catch (Exception ex) {
-                    }
-                });
-                ret = true;
-            } else {
-                ret = FileUtils.writeFile(document.getFile(), document.getContent());
+            try {
+                final byte[] contentAsBytes;
+                if (isEncryptedFile(document.getFile()) && getPassword(shareUtil.getContext()) != null) {
```

Comment:
You have context as function parameter, use that and remove getContext() method from shareutil

Reply:
Good point